Feature/zipped downloads with metadata

agnessa opened the pull request:
https://www.pivotaltracker.com/story/show/159996483

Merge after #681 and at same time as https://github.com/Vizzuality/emissions-scenario-portal/pull/277

aabdaab commented on app/services/api/v1/data/zipped_download.rb:
```diff
@@ -0,0 +1,48 @@
+        def initialize(download_name)
+          @download_name = download_name
+          @temp_dir = "#{Rails.root}/tmp/#{Time.now.strftime('%Y%m%d%H%M%S')}"
+          @data_entries = []
```
why not defining `@data_entries` as attr_accessor along with `@temp_dir`?

agnessa commented on app/services/api/v1/data/metadata_filter.rb:
```diff
@@ -0,0 +1,52 @@
+
+        private
+
+        def select_columns_map
```
Sounds good!

agnessa commented on app/services/api/v1/data/ndc_content_csv_content.rb:
```diff
@@ -4,17 +4,19 @@ module Api
     module Data
       class NdcContentCsvContent
-        def initialize(filter, output)
+        def initialize(filter)
```
I'm not a fan of obscuring constructors, but indeed in a new project with has a lot of initialisers of this could be an elegant approach.

agnessa commented on app/controllers/api/v1/data/historical_emissions_controller.rb:
```diff
@@ -4,6 +4,7 @@ module Data
       class HistoricalEmissionsController < Api::V1::Data::ApiController
         include Streamable
         before_action :parametrise_filter, only: [:index, :download]
+        before_action :parametrise_metadata_filter, only: [:download]
```
Basically I used a filter because there already was one filter in there and I thought it's best to match existing code. Looking at it now I realise it is too long and most of the logic it contains should go inside the filter class. I don't have an issue with filters in general, because they are a Rails classic that most devs are used to, so they're not really surprising. But filters with that much logic in them are a smell, so I'll try to address that.

agnessa commented:
I made some changes to address the comments:
- extracted controller code related to zipping into respective service objects
- extracted select_columns_map into respective modules
- because we ended up with a lot of services and related modules, it made sense to namespace them

I'll apply the same changes to https://github.com/Vizzuality/emissions-scenario-portal/pull/277 (would be good to review & merge both at same time)

agnessa commented:
https://github.com/Vizzuality/emissions-scenario-portal/pull/277 is now ready as well!
